src/js/components/invoice/AddItemsPage.jsx

```diff
@@ -304,21 +310,45 @@ class AddItemsPage extends Component {
       })),
     };
     if (payload.invoiceItems.length) {
-      return apiClient.post(url, payload)
+      return invoiceApi.saveInvoiceItems(this.state.values.id, payload)
         .catch(() => Promise.reject(new Error('react.invoice.error.saveInvoiceItems.label')));
     }
     return Promise.resolve();
   }
 
+  confirmSave(onConfirm) {
+    confirmAlert({
+      title: this.props.translate('react.invoice.message.confirmSave.label', 'Confirm save'),
+      message: this.props.translate(
+        'react.invoice.confirmSave.message',
+        'Are you sure you want to save? There are some lines with empty or zero quantity, those lines will be deleted.',
+      ),
+      buttons: [
+        {
+          label: this.props.translate('react.default.yes.label', 'Yes'),
+          onClick: onConfirm,
+        },
+        {
+          label: this.props.translate('react.default.no.label', 'No'),
+        },
+      ],
+    });
+  }
+
   /**
    * Saves invoices items and goes to the next step.
    * @param {object} values
    * @public
    */
   nextPage(values) {
-    this.saveInvoiceItems(values).then(() => {
-      this.props.nextPage(values);
-    });
+    const hasZeros = _.some(values.invoiceItems, (item) => _.parseInt(item.quantity) === 0);
```

Comment:
I assume that you can't just do:
```
!_.parseInt(item.quantity)
```
right?

Reply:
I would like to avoid this type of an approach like you provided in the comment.
We are not gaining anything besides writing fewer characters in the code.

I strongly believe that this `_.parseInt(item.quantity) === 0` is way more explanatory than this `!_.parseInt(item.quantity)` because in the first case, I explicitly state that I don't want zero values.

My opinion is that this code is much easier to maintain, because when developer reads these lines there is little to think about it, you odn't have to think about casting of bullish values but focus on a concrete case.

As for handling `null` empty strings and `undefined` such logic should be handled  in a form validator method.